AbletonMCP_Remote_Script/__init__.py

```
@@ -507,19 +511,64 @@ def _add_notes_to_clip(self, track_index, clip_index, notes):
                 "note_count": len(notes)
             }
             return result
         except Exception as e:
             self.log_message("Error adding notes to clip: " + str(e))
             raise
+
+    def _get_notes_from_clip(self, track_index, clip_index):
+        """Get all MIDI notes from a clip"""
+        try:
+            if track_index < 0 or track_index >= len(self._song.tracks):
+                raise IndexError("Track index out of range")
```

Ruff (0.14.10)

Abstract `raise` to an inner function

(TRY301)